modules/fedimint-wallet-server/src/lib.rs

```diff
@@ -1305,8 +1289,58 @@ struct StatelessWallet<'a> {
 }
 
 impl<'a> StatelessWallet<'a> {
+    /// Given a tx created from an `WalletOutput`, validate there will be no
+    /// issues submitting the transaction to the Bitcoin network
+    fn validate_tx(
+        &self,
+        tx: &UnsignedTransaction,
+        output: &WalletOutput,
+        consensus_fee_rate: Feerate,
+        network: Network,
+    ) -> Result<(), WalletError> {
+        if let WalletOutput::PegOut(peg_out) = output {
+            if !peg_out.recipient.is_valid_for_network(network) {
+                return Err(WalletError::WrongNetwork(
+                    network,
+                    peg_out.recipient.network,
+                ));
+            }
+        }
+
+        // Validate the tx amount is over the dust limit
+        if tx.peg_out_amount < tx.destination.dust_value() {
+            return Err(WalletError::PegOutUnderDustLimit);
+        }
+
+        // Validate tx fee rate is above the consensus fee rate
+        if tx.fees.fee_rate < consensus_fee_rate {
+            return Err(WalletError::PegOutFeeBelowConsensus(
+                tx.fees.fee_rate.sats_per_kvb,
+                consensus_fee_rate.sats_per_kvb,
+            ));
+        }
+
+        // Validate added fees are above the min relay tx fee
+        let fees = match output {
+            WalletOutput::PegOut(pegout) => pegout.fees.clone(),
+            WalletOutput::Rbf(rbf) => rbf.fees.clone(),
```

Comment:
`rbf.fees` is the *delta* of fees added and we enforce [rule 4 of BIP 125 here](https://github.com/bitcoin/bips/blob/master/bip-0125.mediawiki#implementation-details). Maybe adding some reference to the BIP would be good here for others to know what's going on.

Reply:
Sure I will add a comment